### PR TITLE
[Issue 2482] Migrate delivery metrics transform and load from simpler-grants-sandbox

### PR DESCRIPTION
**Pull request by DavidDudas-Intuitial**
## Summary
Solves for https://github.com/HHS/simpler-grants-gov/issues/2482

## Time to review 
5 mins if you're familiar with the related work in the sandbox repo, otherwise ~15 minutes

## Changes proposed
Imports code from https://github.com/agilesix/simpler-grants-sandbox/tree/main/delivery-metrics. Introduces new db schema, data loader, and metrics calculations scripts for delivery metrics (e.g. percent complete of a deliverable).

## Context for reviewers
Follow instructions in the README.

## Additional information
Output should look similar to this after following README instructions:
```
Davids-MacBook-Pro:delivery-metrics david$ sqlite3 ./db/delivery_metrics.db < ./sql/create_delivery_metrics_db.sql
Davids-MacBook-Pro:delivery-metrics david$ ./src/loader/load_json.py -e 20241007 ./json/example-01.json
...
running data loader with effective date 2024-10-07
opening file '/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/json/example-01.json'
parsing json
found 22 items to process
persisting data
connecting to database 'file:/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/db/delivery_metrics.db?mode=rw'
quad row(s) inserted: 1
quad row(s) updated: 0
deliverable row(s) inserted: 2
deliverable row(s) updated: 0
sprint row(s) inserted: 5
sprint row(s) updated: 0
epic row(s) inserted: 4
epic row(s) updated: 0
issue row(s) inserted: 22
issue row(s) updated: 0
closing db connection
data loader is done
elapsed time: 0.0278 seconds
Davids-MacBook-Pro:delivery-metrics david$ ./src/loader/load_json.py -e 20241014 ./json/example-02.json
...
running data loader with effective date 2024-10-14
opening file '/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/json/example-02.json'
parsing json
found 7 items to process
persisting data
connecting to database 'file:/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/db/delivery_metrics.db?mode=rw'
quad row(s) inserted: 1
quad row(s) updated: 1
deliverable row(s) inserted: 0
deliverable row(s) updated: 1
sprint row(s) inserted: 0
sprint row(s) updated: 2
epic row(s) inserted: 0
epic row(s) updated: 1
issue row(s) inserted: 0
issue row(s) updated: 1
closing db connection
data loader is done
elapsed time: 0.0154 seconds
Davids-MacBook-Pro:delivery-metrics david$ ./src/calculate_percent_complete.py -e 20241010
...
calculating metrics with effective date <= 2024-10-10
verbose mode is OFF
connecting to database 'file:/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/db/delivery_metrics.db?mode=rw'
[QUAD] BY1 Quad One
	[DELIVERABLE] Opportunity listing page (effective 2024-10-07)
		Total Issues: 8
		Total Issues Closed: 0
		Issues Complete: 0.0%
		Total Points: 16
		Total Points Closed: 0
		Points Complete: 0.0%
	[DELIVERABLE] Search Page (effective 2024-10-07)
		Total Issues: 3
		Total Issues Closed: 1
		Issues Complete: 33.3%
		Total Points: 11
		Total Points Closed: 3
		Points Complete: 27.3%
closing db connection
metrics calculations are done
elapsed time: 0.0052 seconds
Davids-MacBook-Pro:delivery-metrics david$ ./src/calculate_percent_complete.py -e 20241015
...
calculating metrics with effective date <= 2024-10-15
verbose mode is OFF
connecting to database 'file:/Users/david/Code/simpler-grants-gov/analytics/delivery-metrics/db/delivery_metrics.db?mode=rw'
[QUAD] BY1 Quad One
	[DELIVERABLE] Opportunity listing page (effective 2024-10-07)
		Total Issues: 8
		Total Issues Closed: 1
		Issues Complete: 12.5%
		Total Points: 16
		Total Points Closed: 5
		Points Complete: 31.2%
	[DELIVERABLE] Search Page (effective 2024-10-14)
		Total Issues: 3
		Total Issues Closed: 1
		Issues Complete: 33.3%
		Total Points: 11
		Total Points Closed: 3
		Points Complete: 27.3%
closing db connection
metrics calculations are done
elapsed time: 0.0051 seconds
```

### CODE REVIEW

**Review by widal001 — CHANGES_REQUESTED** [review]
Let's chat about this more when we connect later today. I should have clarified in the ticket that we'll want to do a bit more to integrate this code with the existing code in `/analytics/src/`

At a minimum we'll want to relocate this to `/analytics/src/` so that we pick up the code quality checks (e.g. type checking, docstrings, test coverage).

Further refactoring is probably out of scope for this ticket, but we should create some follow-up tickets to account for this work once we get a chance to talk through how this new code fits into the existing codebase.

File: analytics/delivery-metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+### Step 1 - Clone the repo
+```
+$ git clone git@github.com:HHS/simpler-grants-gov.git
+```
+
+### Step 2 - Initialize a local instance of the database
+```
+$ cd simpler-grants-gov/analytics/delivery-metrics/
+$ sqlite3 ./db/delivery_metrics.db < ./sql/create_delivery_metrics_db.sql
```
Comment (coilysiren) [inline analytics/delivery-metrics/README.md:13]: This needs to run on postgres

File: analytics/delivery-metrics/src/calculate_percent_complete.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python3
```
Comment (coilysiren) [inline analytics/delivery-metrics/src/calculate_percent_complete.py:1]: This is waaaaaaay too much python to review in a single PR

Comment (coilysiren) [reply analytics/delivery-metrics/src/calculate_percent_complete.py:1]: also it needs unit tests

**Comment by acouch** [comment]
It seems like most of this could be done in Metabase. If we load the project and issue data into the Postgres database we can create dashboards that create deliverable dashboards like the quad-based reports listed above. Has that been tried as a starting point?

**Pull request closed without merging**